src/interfaces.ts

```diff
@@ -143,14 +143,21 @@ export enum FormatEnum {
   PRECOMPUTED = 'PRECOMPUTED',
 }
 
-export interface PrecomputedFlag {
+export type BasePrecomputedFlag = {
   flagKey?: string;
   allocationKey: string;
   variationKey: string;
   variationType: VariationType;
-  variationValue: string;
   extraLogging: Record<string, string>;
   doLog: boolean;
+};
+
+export interface PrecomputedFlag extends BasePrecomputedFlag {
+  variationValue: string;
+}
```

Comment:
major: this type is off from the spec (in terms of optionality of allocation/variation keys and extra logging)

Reply:
done